platform/darwin/test/MGLPredicateTests.mm

```diff
@@ -430,16 +430,16 @@ - (void)testComparisonPredicates {
                           mustRoundTrip:NO];
     }
     {
-        NSArray *expected = @[@"match", @[@"id"], @6, @YES, @5, @YES, @4, @YES, @3, @YES, @NO];
+        NSArray *expected = @[@"match", @[@"id"], @[@"literal", @[@6, @5, @4, @3]], @YES, @NO];
         NSPredicate *predicate = [NSPredicate predicateWithFormat:@"$featureIdentifier IN { 6, 5, 4, 3}"];
         XCTAssertEqualObjects(predicate.mgl_jsonExpressionObject, expected);
-        NSPredicate *predicateAfter = [NSPredicate predicateWithFormat:@"MGL_MATCH(CAST($featureIdentifier, 'NSNumber'), 3, YES, 4, YES, 5, YES, 6, YES, NO) == YES"];
+        NSPredicate *predicateAfter = [NSPredicate predicateWithFormat:@"MGL_MATCH(CAST($featureIdentifier, 'NSNumber'), { 6, 5, 4, 3}, YES, NO) == YES"];
```

Comment:
I’m worried this PR will eventually make it impossible to put an array-typed constant value (for `MGLLineStyleLayer.lineDashPattern`) into a match expression. As I mentioned in https://github.com/mapbox/mapbox-gl-native/issues/11539#issuecomment-387544510, we currently conflate aggregate expressions with array constant values. I think this behavior is desirable; do we need to change it to accommodate this shortcut?

Reply:
Do we agree that the `match` operator has the following syntax:
```
"match", "variable", "valueArrays | keyPath | expression", "trueValue", "defaultValue"
```
The change I am making is that since is possible to have `valueArrays`, the predicate workaround for `IN/CONTAINS` inlines the array as pairs of `value,boolean`, but that is unnecessary because `match` already supports comparing to arrays. This change for `NSPredicate` only "standardizes" the syntax between `NSExpression` and `NSPredicate`. 

What kind of expression are you thinking for `MGLLineStyleLayer.lineDashPattern`?

Reply:
👍 You’re right, I kept conflating input types with output types. Since feature attributes can’t currently be set to arrays, the only obvious use case for an array as a `match` input would be matching return values of a `to-rgba` expression. But even then, I agree that it is understandable that a match expression would only match against atomic values. The only confusion might come from Swift developers used to pattern matching on tuples in `switch` statements.